### DISCUSSION TITLE
Score mutant sequence relative to wt and fine-tune ESM1v with MSA.

### DISCUSSION THREAD

**Discussion by fsee** [facebookresearch/esm#117] [Q&A] [answered]
Hello,

Thanks so much for publishing your awesome models. I was curious what the easiest way would be to score a mutant vs a wt sequence with your zero shot ESM1v method. And then compare that to fine-tuning the model with a specific MSA and performing the same task?

Thanks so much!
Franziska

**Comment by joshim5** [comment, marked as answer]
Hi Franziska,

We provide example code for scoring a mutant vs. a wildtype sequence using the zero-shot ESM-1v method here: https://github.com/facebookresearch/esm/blob/master/variant-prediction/predict.py

The script is setup to accept a wildtype sequence (`"--sequence"`) and a CSV file with the mutations over which you'd like to predict (`"--dms-input"`). If you have just a single mutant sequence, you can either modify the script or use a simple CSV file that contains a single row.

In regards to fine-tuning the model on an MSA, please refer to the general discussion here: https://github.com/facebookresearch/esm/discussions/33

Hope that helps!
Joshua
